### PR TITLE
Complete the docs for Quaternion

### PR DESCRIPTION
**Pull request by HexagonNico**
Added documentation for some methods in the [Quaternion](https://docs.godotengine.org/en/stable/classes/class_quaternion.html) class where it was missing.

### CODE REVIEW

File: doc/classes/Quaternion.xml [hunk 1]
```diff
@@ -79,6 +79,7 @@
 		<method name="exp" qualifiers="const">
 			<return type="Quaternion" />
 			<description>
+				Returns the [url=https://en.wikipedia.org/wiki/Quaternion#Exponential,_logarithm,_and_power_functions]exponential[/url] of this quaternion.
```
Comment (AThousandShips) [inline doc/classes/Quaternion.xml:82]: I'm not sure about linking to Wikipedia here, the link is not guaranteed to be preserved, and we don't do it for the other ones here, I'd suggest providing an explanation here instead, as the equations aren't very complicated

Comment (AThousandShips) [reply doc/classes/Quaternion.xml:82]: Especially considering this is the code, which is trivial to explain:
```
Quaternion Quaternion::log() const {
	Quaternion src = *this;
	Vector3 src_v = src.get_axis() * src.get_angle();
	return Quaternion(src_v.x, src_v.y, src_v.z, 0);
}

Quaternion Quaternion::exp() const {
	Quaternion src = *this;
	Vector3 src_v = Vector3(src.x, src.y, src.z);
	real_t theta = src_v.length();
	src_v = src_v.normalized();
	if (theta < CMP_EPSILON || !src_v.is_normalized()) {
		return Quaternion(0, 0, 0, 1);
	}
	return Quaternion(src_v, theta);
}
```

**Comment by HexagonNico** [comment]
I thought it could be explained with a link since I saw it is done in [Vector3](https://docs.godotengine.org/en/stable/classes/class_vector3.html#class-vector3-method-bezier-interpolate). I changed the docs to explain the code better.

**Review by fire — APPROVED** [review]

**Comment by fire** [comment]
The use of exponential and log of quaternion is that if you randomly pick a value it will be on the rotation sphere, which then you can use then extract into normal quaternion rotation.

**Comment by akien-mga** [comment]
Looks good! Could you squash the commits? See [PR workflow](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) for instructions.

While doing so, please amend the resulting commit message to be more explicit (e.g. like the PR title).

**Comment by HexagonNico** [comment]
Commits squashed!

**Comment by akien-mga** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.2.2.

**Pull request merged**